File: Java/JDK/Java/Java/Java-assigning-class-reference-to-another/CSJava.cs
using System;

//struct Number 
class Number 
{
    public int x;
}

class CSJava 
{
    void Print()
    {
	Number a = new Number();
        Number b = new Number();

	a.x = 10;
	b.x = 20;

	Console.WriteLine("a.x {0}, b.x {1}", a.x, b.x);

	a = b;
	b.x = 30;

	Console.WriteLine("a.x {0}, b.x {1}", a.x, b.x);
    }

    static void Main()
    {
	CSJava cj = new CSJava();
        cj.Print();
    }
}

File: Java/JDK/Java/Java/Java-assigning-class-reference-to-another/CSJava.java
import static java.lang.System.out;

final class Number {
    public int x;
}

class CSJava {
    void print() {
	Number a = new Number();
        Number b = new Number();

	a.x = 10;
	b.x = 20;

	out.format("a.x %d, b.x %d", a.x, b.x);

	out.println();

	a = b;
	b.x = 30;

	out.format("a.x %d, b.x %d", a.x, b.x);
    }

    public static void main(String[] args) {
	CSJava cj = new CSJava();
        cj.print();
    }
}